pkg/controller/podautoscaler/horizontal.go

```diff
@@ -652,7 +652,9 @@ func (a *HorizontalController) reconcileAutoscaler(hpav1Shared *autoscalingv1.Ho
 		var metricTimestamp time.Time
 		metricDesiredReplicas, metricName, metricStatuses, metricTimestamp, err = a.computeReplicasForMetrics(hpa, scale, hpa.Spec.Metrics)
 		if err != nil {
-			a.setCurrentReplicasInStatus(hpa, currentReplicas)
+			// set desiredReplicas to currentReplicas when can't compute metrics
+			desiredReplicas = currentReplicas
```

Comment:
This is not always correct.  E.g. when a Deployment is rolling out with surge pods configured there will always be more pods than desired.  Desired replicas should reflect what the HPA recommends / would recommend (scale subresource.spec) not how many are observed (scale subresource.status).  Writing data from status back to spec is the root cause of a positive feedback loop (uncontrolled scale up): https://github.com/kubernetes/kubernetes/pull/79035

If you want to write something to desiredReplicas, it should be the scale subresource replicas value.

Reply:
@josephburnett    thx! You are correct. I change desiredReplicas to scale.Spec.Replicas and pushed code again. Please take time to look.thx again!